### PR TITLE
Issue #PS-891 feat: Fetching facilitator list and their respective centers and added pop up

### PR DESCRIPTION
**Pull request by suvarnakale**
…nters, and added popup

## Description

Please describe the changes in your pull request in few words here.

### Changes

List the changes done to fix a bug or introducing a new feature.

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

## Summary by CodeRabbit

- **New Features**
  - Added functionality to fetch and display facilitator and cohort data.
  - Introduced user management with modal components for detailed user information.
  
- **Bug Fixes**
  - Improved data handling and error management for fetching facilitator lists.

- **Refactor**
  - Streamlined state management for user and cohort data.
  - Enhanced modularity by refactoring data fetching into a separate service.

### CODE REVIEW

File: src/pages/manageUser.tsx [hunk 1]
```diff
@@ -6,24 +7,139 @@ import InputAdornment from '@mui/material/InputAdornment';
...
+
+  useEffect(() => {
+    const getFacilitator = async () => {
+      try {
+        let state, district;
+        if (typeof window !== 'undefined' && window.localStorage) {
+          state = localStorage.getItem('state');
+          district = localStorage.getItem('district');
+        }
+        if (state && district) {
```
Comment (itsvick) [inline src/pages/manageUser.tsx:58]: what if state and district is not available??

File: src/pages/manageUser.tsx [hunk 2]
```diff
@@ -6,24 +7,139 @@ import InputAdornment from '@mui/material/InputAdornment';
...
+          const extractedData = resp.map((user: any) => ({
+            userId: user.userId,
+            name: user.name,
+          }));
+          setTimeout(() => {
+            setUsers(extractedData);
+          });
+        }
+      } catch (error) {
+        console.log(error);
```
Comment (itsvick) [inline src/pages/manageUser.tsx:78]: Show some error toast message

File: src/pages/manageUser.tsx [hunk 3]
```diff
@@ -6,24 +7,139 @@ import InputAdornment from '@mui/material/InputAdornment';
...
+              return acc;
+            },
+            {}
+          );
+          console.log('allCohortsData', allCohortsData);
+
+          setCohortsData(allCohortsData);
+        }
+      } catch (error) {
+        console.log(error);
```
Comment (itsvick) [inline src/pages/manageUser.tsx:117]: show the error message

File: src/services/ManageUser.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+export const getFacilitorList = async ({
+  limit,
+  page,
+  filters,
+}: FacilitatorListParam): Promise<any> => {
+  const apiUrl: string = `${process.env.NEXT_PUBLIC_BASE_URL}/list`;
+  try {
+    const response = await post(apiUrl, { limit, page, filters });
+    return response?.data?.result;
+  } catch (error) {
```
Comment (itsvick) [inline src/services/ManageUser.ts:13]: Add error toast message

File: src/services/ManageUser.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { FacilitatorListParam } from '@/utils/Interfaces';
+import { post } from './RestClient';
+
+export const getFacilitorList = async ({
```
Comment (itsvick) [inline src/services/ManageUser.ts:4]: Correct the spelling mistake

**Review by itsvick — CHANGES_REQUESTED** [review]

File: src/pages/manageUser.tsx (hunk 1 above)
Comment (suvarnakale) [reply src/pages/manageUser.tsx:58]: API call will not happen

File: src/pages/manageUser.tsx (hunk 2 above)
Comment (suvarnakale) [reply src/pages/manageUser.tsx:78]: okay

File: src/services/ManageUser.ts (hunk 5 above)
Comment (suvarnakale) [reply src/services/ManageUser.ts:4]: ok

File: src/services/ManageUser.ts (hunk 4 above)
Comment (suvarnakale) [reply src/services/ManageUser.ts:13]: this is service, we dont need toast msg here.

File: src/pages/manageUser.tsx (hunk 3 above)
Comment (suvarnakale) [reply src/pages/manageUser.tsx:117]: yes

**Pull request merged**